### PR TITLE
feat: add support for listening pending interactions

### PR DESCRIPTION
**Pull request by sarvalabs-harshrastogi**
This pull request enables to listen to pending interactions

- closes: #16 

#### Changes

- Implemented a WebSocket event, `new_pending_interactions`, enabling clients to subscribe and receive real-time pending interaction hash through this channel.

# Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist
- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)

### CODE REVIEW

File: packages/js-moi-providers/__tests__/ws-provider.test.ts
Comment (sarvalabs-gokul) [inline packages/js-moi-providers/__tests__/ws-provider.test.ts:1]: The initial test case should be configured to listen for the `ALL_TESSERACTS` event; however, it is currently set to listen to an event that returns tesseracts specific to an account.

File: docs/source/providers.rst [hunk 1]
```diff
@@ -572,6 +572,9 @@ height, hash, timestamp, and other relevant data.
 the given address is mined on the blockchain. It provides information about the 
 tesseract.
 
+``PENDING_INTERACTIONS`` - This event is triggered when a new pending interaction
+occurs in the blockchain, providing a unique hash for the interaction.
```
Comment (sarvalabs-gokul) [inline docs/source/providers.rst:576]: can be improved.

**Review by sarvalabs-gokul — CHANGES_REQUESTED** [review]

File: docs/source/providers.rst (hunk 1 above)
Comment (sarvalabs-harshrastogi) [reply docs/source/providers.rst]: 25696762c8174ea9b4c5d3bab8ce0863d2d3cb5c

File: packages/js-moi-providers/__tests__/ws-provider.test.ts
Comment (sarvalabs-harshrastogi) [reply packages/js-moi-providers/__tests__/ws-provider.test.ts:1]: 25696762c8174ea9b4c5d3bab8ce0863d2d3cb5c

**Review by sarvalabs-gokul — APPROVED** [review]

**Review by sarvalabs-gokul — APPROVED** [review]

**Pull request merged**